openwisp_monitoring/device/base/models.py

```diff
@@ -454,6 +454,27 @@ def handle_disabled_organization(cls, organization_id):
             status='unknown'
         )
 
+    @classmethod
+    def handle_critical_metric(cls, check, **kwargs):
+        """
+        Updates the device status to 'unknown' if the given check
+        is for a critical metric from the CRITICAL_DEVICE_METRICS setting
+        and the check is not active.
+        """
+        from django.conf import settings
+
+        critical_metrics = getattr(
+            settings,
+            'OPENWISP_MONITORING_CRITICAL_DEVICE_METRICS',
+            [{'key': 'ping', 'field_name': 'reachable'}],
+        )
+        if check.check_type in [m['key'] for m in critical_metrics]:
```

Comment:
The `check_type` value for ping checks is `openwisp_monitoring.check.classes.Ping`, so this if block will never be executed with this code.

We can get the critical metrics like:

```python
Metric.objects.filter(key__in=critical_metric_list, object_id=check.object_id, content_type=check.content_type)
```

But you need to build `critical_metric_list` by preprocessing the output of `get_critical_device_metrics` so the result is a list of strings (now it returns a list of dicts and it won't work).